tck/features/expressions/pattern/Pattern2.feature

```diff
@@ -28,5 +28,211 @@
 
 #encoding: utf-8
 
-Feature: Pattern2 - Number of Pattern Matches
+Feature: Pattern2 - Pattern Comprehension
 
+  Scenario: [1] Returning a pattern comprehension
+    Given an empty graph
+    And having executed:
+      """
+      CREATE (a:A)
+      CREATE (a)-[:T]->(:B),
+             (a)-[:T]->(:C)
+      """
+    When executing query:
+      """
+      MATCH (n)
+      RETURN [p = (n)-->() | p] AS ps
+      """
+    Then the result should be, in any order:
+      | ps                                     |
+      | [<(:A)-[:T]->(:C)>, <(:A)-[:T]->(:B)>] |
+      | []                                     |
+      | []                                     |
+    And no side effects
+
+  Scenario: [2] Returning a pattern comprehension with label predicate
+    Given an empty graph
+    And having executed:
+      """
+      CREATE (a:A), (b:B), (c:C), (d:D)
+      CREATE (a)-[:T]->(b),
+             (a)-[:T]->(c),
+             (a)-[:T]->(d)
+      """
+    When executing query:
+      """
+      MATCH (n:A)
+      RETURN [p = (n)-->(:B) | p]
+      """
+    Then the result should be, in any order:
+      | [p = (n)-->(:B) \| p] |
+      | [<(:A)-[:T]->(:B)>]   |
+    And no side effects
+
+  Scenario: [3] Returning a pattern comprehension with bound nodes
+    Given an empty graph
+    And having executed:
+      """
+      CREATE (a:A), (b:B)
+      CREATE (a)-[:T]->(b)
+      """
+    When executing query:
+      """
+      MATCH (a:A), (b:B)
+      RETURN [p = (a)-[*]->(b) | p] AS paths
+      """
+    Then the result should be, in any order:
+      | paths               |
+      | [<(:A)-[:T]->(:B)>] |
+    And no side effects
+
+  Scenario: [4] Introducing new node variable in pattern comprehension
+    Given an empty graph
+    And having executed:
+      """
+      CREATE (a), (b {name: 'val'})
+      CREATE (a)-[:T]->(b)
```

Comment:
not a comment on this PR per se, but I'm curious if you have some guideline towards the formatting of setup queries?

Concretely, I'm thinking whether there is some stylistic rule that dictates the form you chose rather than these equivalent forms:

```
CREATE (a)-[:T]->(b {name: 'val'})
```
```
CREATE (a), (b {name: 'val'}),
       (a)-[:T]->(b)
```